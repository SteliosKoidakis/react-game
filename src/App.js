import React from 'react';
import { MemoryGameView } from 'src/views';
import { API_URL, CARD_NUMBERS } from 'src/constants';
import './App.scss';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          Github Memory
        </p>
      </header>
      <main>
        <MemoryGameView
          url={API_URL}
          cardNumbers={CARD_NUMBERS}
        />
      </main>
    </div>
  );
}

export default App;
