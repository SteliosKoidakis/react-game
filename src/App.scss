
.App {
  padding: 0;

  @media (min-width: 600px) {
    padding: 10px;
  }

  &__header {
    text-align: center;
    padding: 0;

    @media (min-width: 600px) {
      padding-bottom: 10px;
    }
  }
}