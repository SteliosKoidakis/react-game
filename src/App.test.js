import React from 'react';
import { render } from '@testing-library/react';

import App from './App';

describe('Given App component', () => {
  describe('When has been called', () => {
    it('Then it should render MemoryGameView component', () => {
      const { getByTestId } = render(<App />);

      expect(getByTestId('memoryGameView')).toBeInTheDocument();
    });
  });
});
