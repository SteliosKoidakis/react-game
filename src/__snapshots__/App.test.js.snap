// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given component App Then it should match snapshot output 1`] = `
"<div className=\\"App\\">
  <header className=\\"App-header\\">
    <p>
      Github Memory
    </p>
  </header>
  <main>
    <MemoryGameView url=\\"https://api.github.com/repos/facebook/react/contributors?\\" cardNumbers={6} />
  </main>
</div>"
`;
